### PR TITLE
Final client changes for TreeEntry.Commit and ISnapshotTree.commits removal

### LINKED ISSUE #4691: Get rid of commits in runtime code
It would be great to get rid of these concepts from repo:

TreeEntry.Commit
"commit"
ISnapshotTree.commits

Notion of commits in summaries is not used and proposed to be cleaned up here: https://github.com/microsoft/FluidFramework/pull/4682

### PR DESCRIPTION
**Pull request by NicholasCouri**
Fixes #4691 - After the prerelease of server/routerlicious , common/lib/protocol-definitions, common/lib/container-definitions and common/lib/driver-definitions, we can finally incorporate the new prerelease numbers and remove the last references of TreeEntry.Commit and ISnapshotTree.commits

### CODE REVIEW

File: packages/dds/register-collection/src/test/consensusRegisterCollection.spec.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ describe("ConsensusRegisterCollection", () => {
             });
 
             it("load", async () => {
-                const tree: ITree = buildTree(expectedSerialization);
+                const tree: ITree = buildTree(expectedSerialization) as ITree;
```
Comment (vladsud) [inline packages/dds/register-collection/src/test/consensusRegisterCollection.spec.ts:145]: Curious, why this cast is required?

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 2]
```diff
@@ -356,38 +356,20 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
...
 
-        for (const key of Object.keys(snapshotTree.commits)) {
-            commits[decodeURIComponent(key)] = snapshotTree.commits[key];
-        }
-
-        let finalTree: api.ISnapshotTree;
-        // For container loaded from detach new summary, we will not have a commit for ".app" in downloaded summary as the client uploaded both
-        // ".app" and ".protocol" trees by itself. For other summaries, we will have ".app" as commit because client previously only uploaded the
-        // app summary.
-        if (commits && commits[".app"]) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:370]: Can you please run it by Jatin and maybe Marcelo?
SPO for some time used commits. I think it's gone, but I'm not 100% sure.
If this flow is still needed, we can define SPO specific version of ISnapshotTree - it should be local only in this file as outside of the SPO driver we expose proper ISnapshotTree without commits (i.e. this code was flattening it in the past)

Comment (vladsud) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:370]: @jatgarg - please take a look

File: packages/drivers/odsp-driver/src/odspSnapshotParser.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ import { ISnapshotContents } from "./odspUtils";
 function buildHierarchy(flatTree: IOdspSnapshotCommit): api.ISnapshotTree {
     const lookup: { [path: string]: api.ISnapshotTree } = {};
     // id is required for root tree as it will be used to determine the version we loaded from.
-    const root: api.ISnapshotTree = { id: flatTree.id, blobs: {}, commits: {}, trees: {} };
+    const root: api.ISnapshotTree = { id: flatTree.id, blobs: {}, trees: {} };
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspSnapshotParser.ts:21]: Do we use ID anywhere? I think ISnapshotTree.id should be deleted as runtime has no capability to leverage it (there is no API to do anything with it).

File: packages/loader/driver-utils/src/treeConversions.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ export function convertSummaryTreeToSnapshotITree(
                     parsedContent = Uint8ArrayToString(value.content, "base64");
                     encoding = "base64";
                 }
-                entries.push(new BlobTreeEntry(k, parsedContent, encoding));
+                entries.push(new BlobTreeEntry(k, parsedContent, encoding) as ITreeEntry);
```
Comment (vladsud) [inline packages/loader/driver-utils/src/treeConversions.ts:55]: Similar question - why do we need a cast?

File: packages/dds/register-collection/src/test/consensusRegisterCollection.spec.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/dds/register-collection/src/test/consensusRegisterCollection.spec.ts:145]: Not needed - figured out it was a protocol-base mismatch. Fixed in the latest iteration.

File: packages/drivers/odsp-driver/src/odspSnapshotParser.ts (hunk 3 above)
Comment (NicholasCouri) [reply packages/drivers/odsp-driver/src/odspSnapshotParser.ts:21]: We do use the ID still.

File: packages/loader/driver-utils/src/treeConversions.ts (hunk 4 above)
Comment (NicholasCouri) [reply packages/loader/driver-utils/src/treeConversions.ts:55]: as before, I just reverted it - it was hte protocol-base mismatch.

File: packages/tools/fetch-tool/src/fluidFetchSnapshot.ts [hunk 5]
```diff
@@ -115,8 +115,6 @@ async function fetchBlobsFromSnapshotTree(
     tree: ISnapshotTree,
     prefix: string = "/",
     perCommitBlobIdMap?: Map<string, number>): Promise<IFetchedData[]> {
```
Comment (scarlettjlee) [inline packages/tools/fetch-tool/src/fluidFetchSnapshot.ts:117]: perCommitBlobIdMap and const commit below should be renamed. The difference now is subtree versus top-level.

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts (hunk 2 above)
Comment (NicholasCouri) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:370]: @marcmasmsft @znewton  can you confirm whether we still need the commit on the SPO ? We don't think so but want to confirm. Thanks!

Comment (NicholasCouri) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:370]: Marcelo just confirmed they do not use commit anymore.

**Review by marcmasmsft — APPROVED** [review]

**Comment by heliocliu** [comment]
can you include the breaking.md entry now that the overall change has bubbled up into the client layer?

**Comment by NicholasCouri** [comment]
sure thing.

---
In reply to: [1082413562](https://github.com/microsoft/FluidFramework/pull/9633#issuecomment-1082413562) [](http://example.com/codeflow?ancestors=1082413562,1082413562)

**Review by scarlettjlee — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts (hunk 2 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:370]: Yeah we don't use it anymore. We can remove it.

**Review by jatgarg — APPROVED** [review]

**Pull request merged**